backend/tests/slack/commands/gsoc_tests.py

```diff
@@ -41,9 +47,46 @@ def test_handler_responses(
         ):
             handler(ack=MagicMock(), command=command, client=mock_slack_client)
 
-        if not commands_enabled:
-            mock_slack_client.conversations_open.assert_not_called()
-            mock_slack_client.chat_postMessage.assert_not_called()
-        else:
+            if not commands_enabled:
+                mock_slack_client.conversations_open.assert_not_called()
+                mock_slack_client.chat_postMessage.assert_not_called()
+            else:
+                mock_slack_client.conversations_open.assert_called_once_with(
+                    users=command["user_id"]
+                )
+                mock_slack_client.chat_postMessage.assert_called_once()
+
+                blocks = mock_slack_client.chat_postMessage.call_args[1]["blocks"]
+                block_texts = [block["text"]["text"] for block in blocks]
+
+                if command_text == "":
+                    assert "GSOC_GENERAL_INFORMATION_BLOCKS" in block_texts[0]
+                    assert FEEDBACK_CHANNEL_MESSAGE in block_texts[1]
+                else:
+                    assert any(expected_message in text for text in block_texts)
+
+    def test_handler_with_projects(self, mock_slack_client):
+        """Test handler when projects are found for a valid year."""
+        mock_projects = {
+            "hits": [
+                {
+                    "idx_name": "Test Project",
+                    "idx_url": "https://owasp.org/www-project-bug-logging-tool/",
+                }
+            ]
+        }
```

Comment:
And this should be changed to:
```suggestion
        mock_project = Mock()
        mock_project.nest_url = "https://owasp.org/www-project-bug-logging-tool/"
        mock_project.owasp_name = "Test Project"
        mock_projects = [mock_project]

```